tests/governance.py

```diff
@@ -627,6 +627,165 @@ def test_cose_auth(network, args):
             headers={"content-type": "application/cose"},
         )
         assert r.status_code == 401
+    return network
+
+
+@reqs.description("Test COSE ack")
+def test_cose_ack(network, args):
+    primary, _ = network.find_primary()
+    identity = network.identity("member0")
+    signed_statement = signing.create_cose_sign1(
+        b"",
+        open(identity.key, encoding="utf-8").read(),
+        open(identity.cert, encoding="utf-8").read(),
+        {"ccf.gov.msg.type": "state_digest"},
+    )
+    with primary.client() as c:
+        r = c.post(
+            "/gov/ack/update_state_digest",
+            body=signed_statement,
+            headers={"content-type": "application/cose"},
+        )
+        assert r.status_code == 200
+
+    signed_state_digest = signing.create_cose_sign1(
+        r.body.data(),
+        open(identity.key, encoding="utf-8").read(),
+        open(identity.cert, encoding="utf-8").read(),
+        {"ccf.gov.msg.type": "ack"},
+    )
+    with primary.client() as c:
+        r = c.post(
+            "/gov/ack",
+            body=signed_state_digest,
+            headers={"content-type": "application/cose"},
+        )
+        assert r.status_code == 204
+    return network
+
+
+@reqs.description("Test COSE proposal")
+def test_cose_proposal(network, args):
+    primary, _ = network.find_primary()
+    identity = network.identity("member0")
+    other_identity = network.identity("member1")
+
+    new_user_local_id = "alice"
+    new_user = network.create_user(new_user_local_id, args.participants_curve)
+    template_loader = jinja2.ChoiceLoader(
+        [
+            jinja2.FileSystemLoader(args.jinja_templates_path),
+            jinja2.FileSystemLoader(os.path.dirname(new_user.cert_path)),
+        ]
+    )
+    template_env = jinja2.Environment(
+        loader=template_loader, undefined=jinja2.StrictUndefined
+    )
+    proposal_template = template_env.get_template("set_user_proposal.json.jinja")
+    proposal_body = proposal_template.render(cert=os.path.basename(new_user.cert_path))
```

Comment:
To avoid introducing this Jinja arg dependency on all the suites that want to run this test, we could use `consortium.make_proposal` here. I think that always tries to dump the proposals to file rather than returning the bodies, but we can separate out the method to do the first and use it here.

Reply:
Agreed, but I think governance, governance_js and governance_history are due for a proper merge & cleanup, so I will focus on that.